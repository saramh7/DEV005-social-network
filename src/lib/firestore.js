/**
 * En este archivo tiene todas las funciones que conectar√°n el DOM con Firestore.
 */
