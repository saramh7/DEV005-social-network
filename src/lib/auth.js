import {
  GoogleAuthProvider, signInWithPopup, signOut,
} from 'firebase/auth';
import { auth } from './firebase.js';

const provider = new GoogleAuthProvider();
auth.languageCode = 'es';

export const googleLogin = async () => {
  await signInWithPopup(auth, provider)
    .then((result) => {
      // This gives you a Google Access Token. You can use it to access the Google API.
      const credential = GoogleAuthProvider.credentialFromResult(result);
      const token = credential.accessToken;
      // The signed-in user info.
      const user = result.user;
      // IdP data available using getAdditionalUserInfo(result)

      console.log('Usuario Loggeado exitosamente', result);
    }).catch((error) => {
      // Handle Errors here.
      const errorCode = error.code;
      const errorMessage = error.message;
      // The email of the user's account used.
      const email = error.customData.email;
      // The AuthCredential type that was used.
      const credential = GoogleAuthProvider.credentialFromError(error);
      // ...

      console.log('Error', error);
    });
};

export const googleLogout = async () => {
  await signOut(auth).then(() => {
    console.log('Sign-out successful');
    // Sign-out successful.
  }).catch((error) => {
    console.log('Error', error);
    // An error happened.
  });
};
