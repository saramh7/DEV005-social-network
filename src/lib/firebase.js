// Import the functions you need from the SDKs you need
import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getFirestore } from 'firebase/firestore';
import { getAnalytics } from 'firebase/analytics';
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: 'AIzaSyAz6OSQ_9gp07rKjoq31lfRM0zbjGBzbYU',
  authDomain: 'social-network-te-recomiendo.firebaseapp.com',
  projectId: 'social-network-te-recomiendo',
  storageBucket: 'social-network-te-recomiendo.appspot.com',
  messagingSenderId: '647650903243',
  appId: '1:647650903243:web:6da88eb1edc7230645c74e',
  measurementId: 'G-GTG1855LG3',
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
const auth = getAuth(app);
const analytics = getAnalytics(app);
const db = getFirestore(app);

export { auth, db, analytics };
