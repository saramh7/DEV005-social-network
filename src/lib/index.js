/**
 * En este archivo se cargan todas las configuraciones para utilizar los servicios
 * de firebase. Se exportan las variables auth, db y analytics que contienen las
 * instancias a cada servicio para ser utilizadas en los archivos correspondientes
 * a cada servicio.
 *
 * const auth = contiene la instancia para el servicio de autenticaci贸n
 * const db = contiene la instancia para el servicio de firestore (base de datos)
 * const analytics = contiene la instancia para el servicio de google analytics
 *
 * Este archivo se construy贸 con la configuraci贸n propuesta por la documentaci贸n
 * de Firebase. https://firebase.google.com/docs/web/setup?authuser=0&hl=es
 * NO MODIFICAR
 */

// Import the functions you need from the SDKs you need
import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getFirestore } from 'firebase/firestore';
// import { getAnalytics } from 'firebase/analytics';

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: 'AIzaSyAz6OSQ_9gp07rKjoq31lfRM0zbjGBzbYU',
  authDomain: 'social-network-te-recomiendo.firebaseapp.com',
  projectId: 'social-network-te-recomiendo',
  storageBucket: 'social-network-te-recomiendo.appspot.com',
  messagingSenderId: '647650903243',
  appId: '1:647650903243:web:6da88eb1edc7230645c74e',
  measurementId: 'G-GTG1855LG3',
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
const auth = getAuth(app);
// const analytics = getAnalytics(app);
const db = getFirestore(app);

export { auth, db };
