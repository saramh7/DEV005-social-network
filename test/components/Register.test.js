/**
 * @jest-environment jsdom
 */

import { signInNewAccount } from '../../src/lib/auth';

describe('signInNewAccount', () => {
  test('it should to be a function', () => {
    expect(typeof signInNewAccount).toBe('function');
  });

  test('La función signInNewAccount permite el acceso con una cuenta de correo válida', async () => {
    const email = 'prueba@prueba.com';
    const password = '123456';

    await expect(signInNewAccount(email, password))
      .resolves.not.toThrow();

    // verificar que el usuario está firmado en
    const user = auth.currentUser;
    expect(user).not.toBeNull();
    expect(user.email).toBe(email);
  });

  test('snapshot of signInNewAccount', () => {
    const template = document.createElement('div');
    template.append(signInNewAccount());
    expect(template).toMatchSnapshot();
  });
});
