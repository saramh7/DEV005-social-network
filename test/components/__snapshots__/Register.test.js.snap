// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signInNewAccount snapshot of signInNewAccount 1`] = `
<div>
  [object Promise]
</div>
`;
